core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/jdbc/JDBCUserStoreManager.java

```diff
@@ -3743,19 +3748,49 @@ protected PaginatedSearchResult doGetUserList(Condition condition, String profil
             }
 
             SqlBuilder sqlBuilder = getQueryString(isGroupFiltering, isUsernameFiltering, isClaimFiltering,
-                    expressionConditions, limit, offset, sortBy, sortOrder, profileName, type);
-            prepStmt = dbConnection.prepareStatement(sqlBuilder.getQuery());
-            populatePrepareStatement(sqlBuilder, prepStmt);
-            rs = prepStmt.executeQuery();
-            while (rs.next()) {
-                String name = rs.getString(1);
-                list.add(UserCoreUtil.addDomainToName(name, getMyDomainName()));
+                    expressionConditions, limit, offset, sortBy, sortOrder, profileName, type, totalMultiGroupFilters,
+                    totalMulitClaimFitlers);
+
+            if (MYSQL.equals(type) && totalMultiGroupFilters > 1 && totalMulitClaimFitlers > 1) {
+                String fullQuery = sqlBuilder.getQuery();
+                String[] splits = fullQuery.split("INTERSECT ");
+                int startIndex = 0;
+                int endIndex = 0;
+                for (String query : splits) {
+                    List<String> tempUserList = new ArrayList<>();
+                    int occurance = StringUtils.countMatches(query, "?");
```

Comment:
shall we make "?" a constant

Reply:
Addressed with https://github.com/wso2/carbon-kernel/pull/2272